presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraTokenSplitManager.java

```diff
@@ -65,8 +69,25 @@ public void testNonEmptyTable()
             session.execute(format("INSERT INTO %s.%s (key) VALUES ('%s')", KEYSPACE, tableName, "value" + i));
         }
         EmbeddedCassandra.refreshSizeEstimates(KEYSPACE, tableName);
-        List<TokenSplit> splits = splitManager.getSplits(KEYSPACE, tableName);
-        assertEquals(splits.size(), PARTITION_COUNT / SPLIT_SIZE);
+        checkSplits(tableName, PARTITION_COUNT / SPLIT_SIZE);
         session.execute(format("DROP TABLE %s.%s", KEYSPACE, tableName));
     }
+
+    private void checkSplits(String tableName, int splitSize)
+    {
+        long start = System.nanoTime();
+        while (!Thread.currentThread().isInterrupted()) {
+            try {
+                List<TokenSplit> splits = splitManager.getSplits(KEYSPACE, tableName);
+                assertEquals(splits.size(), splitSize);
+                return;
+            }
+            catch (AssertionError e) {
```

Comment:
I agree with the other comments, but wanted to point out that catching an exception that you explicitly throw (via `assertEquals`) should be avoided. This retry loop could be written to loop until time expires or the count correct.